### PR TITLE
REFACTOR: admin users controller specs to requests

### PR DESCRIPTION
**Pull request by OsamaSayegh**

### CODE REVIEW

File: spec/requests/admin/users_controller_spec.rb [hunk 1]
```diff
@@ -4,6 +4,810 @@
...
+  describe '#index' do
+    it 'returns success with JSON' do
+      get "/admin/users/list.json"
+      expect(response.status).to eq(200)
+      expect(JSON.parse(response.body)).to be_present
+    end
+
+    context 'when showing emails' do
+      it "returns email for all the users" do
+        get "/admin/users/list.json", params: { show_emails: "true" }
```
Comment (tgxworld) [inline spec/requests/admin/users_controller_spec.rb]: We're missing a response code here.

File: spec/requests/admin/users_controller_spec.rb [hunk 2]
```diff
@@ -4,6 +4,810 @@
...
+        data = ::JSON.parse(response.body)
+        data.each do |user|
+          expect(user["email"]).to be_present
+        end
+      end
+
+      it "logs only 1 enty" do
+        expect(UserHistory.where(action: UserHistory.actions[:check_email], acting_user_id: admin.id).count).to eq(0)
+
+        get "/admin/users/list.json", params: { show_emails: "true" }
```
Comment (tgxworld) [inline spec/requests/admin/users_controller_spec.rb]: Missing response code :)

File: spec/requests/admin/users_controller_spec.rb [hunk 3]
```diff
@@ -4,6 +4,810 @@
...
+
+        expect(UserHistory.where(action: UserHistory.actions[:check_email], acting_user_id: admin.id).count).to eq(1)
```
Comment (tgxworld) [inline spec/requests/admin/users_controller_spec.rb]: There is no need to duplicate this line twice. We can do `expect {}.to change {}.by(...)`

File: spec/requests/admin/users_controller_spec.rb [hunk 4]
```diff
@@ -4,6 +4,810 @@
...
+    context 'an existing user' do
+      it 'returns success' do
+        get "/admin/users/#{user.id}.json"
+        expect(response.status).to eq(200)
+      end
+    end
+
+    context 'a non-existing user' do
+      it 'returns 404 error' do
+        get "/admin/users/0.json", params: { id: 0 }
```
Comment (tgxworld) [inline spec/requests/admin/users_controller_spec.rb]: I don't think the `id` params is required here.

File: spec/requests/admin/users_controller_spec.rb [hunk 5]
```diff
@@ -4,6 +4,810 @@
...
+
+      expect(response.status).to eq(200)
+
+      user.reload
+      expect(user).to be_suspended
+      expect(user.suspended_at).to be_present
+      expect(user.suspended_till).to be_present
+
+      log = UserHistory.where(target_user_id: user.id).order('id desc').first
+      expect(log).to be_present
```
Comment (tgxworld) [inline spec/requests/admin/users_controller_spec.rb]: this line is not required

File: spec/requests/admin/users_controller_spec.rb [hunk 6]
```diff
@@ -4,6 +4,810 @@
...
+    end
+
+    context "with an associated post" do
+      it "can have an associated post" do
+        put "/admin/users/#{user.id}/suspend.json", params: suspend_params
+
+        expect(response.status).to eq(200)
+
+        log = UserHistory.where(target_user_id: user.id).order('id desc').first
+        expect(log).to be_present
```
Comment (tgxworld) [inline spec/requests/admin/users_controller_spec.rb]: this line is not required

**Comment by OsamaSayegh** [comment]
@tgxworld done ✅

**Pull request merged**